common/src/main/java/com/hortonworks/streamline/common/SchemaValueConverter.java

```diff
@@ -82,10 +85,7 @@ private SchemaValueConverter() {
      * @return the converted value
      */
     public static Object convert(Schema.Type type, Object value) {
-        if (value == null) {
-            return null;
-        }
-
+        Objects.requireNonNull(value, "'null' value is not allowed");
```

Comment:
We may want to handle this to IllegalArgumentException, and change above `forEach` to catch IllegalArgumentException and wrap with SchemaValidationFailedException. 
Bubbling up NullPointerException or IllegalArgumentException to the REST API response would be bad for end users.

If you just want to ensure denying null, I'm OK to leave it as it is, and I could handle it.

Reply:
I reverted the changes in `convert` and added a check in the `foreach` to throw `SchemaValidationException` if value is null.